pkg/controllers/deprovisioning/validation.go

```diff
@@ -97,6 +101,30 @@ func (v *Validation) IsValid(ctx context.Context, cmd Command) (bool, error) {
 	return isValid, nil
 }
 
+// Returns if the candidates chosen can still be terminated after the TTL
+func (v *Validation) validateCandidates(cmd Command, pdbs *PDBLimits) bool {
+	for _, n := range cmd.candidates {
+		if !n.Node.DeletionTimestamp.IsZero() {
```

Comment:
This makes sense to me to emit an event. We do filter these inside GetCandidates(), but that grabs the whole state of the cluster and recomputes the list of candidate nodes, which is more expensive than just checking the existing list. 

If I have a node that is terminating, I'd like to know that Consolidation is aware of it, but intentionally waiting for it to finish. It'll return some events, in this case here, it'll be a node that wasn't terminating before the TTL, then started terminating after the TTL. This can impact the Consolidation decision as well, as there might be a better choice for Consolidation now to reduce pod churn.

Reply:
Why should consolidation be aware of a terminating the node? That node will *never* be a candidate for consolidation.  Seems like it should be safe to ignore it and move on.

Reply:
If we know that the node is being terminated, the previous scheduling decision we made that returned a list of replacement nodes might be incorrect now, or even less optimal. We should let the node terminate and the pods evict before we try to make another consolidation action as to reduce churn or take a consolidation decision with stale/incorrect information.

Reply:
Gotcha -- was missing some context. Reading more of the code to understand this better. Looks like this code is essentially copied in consolidation.go:83